components/NewRealmWizard/components/steps/AddNFTCollectionForm.tsx

```diff
@@ -73,36 +90,64 @@ async function enrichCollectionInfo(connection, collectionKey) {
   )
 }
 
-async function getNFTCollectionInfo(connection, collectionKey) {
-  const { data: result } = await deprecated.Metadata.findByMint(
-    connection,
-    collectionKey
-  )
-  console.log('NFT findByMint result', result)
-  if (result?.collection?.verified && result.collection?.key) {
+async function getNFTCollectionInfo(
+  connection: Connection,
+  collectionKey: string
+) {
+  const result = await fetch('https://graph.holaplex.com/v1', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      query: getNftQuery(1, 0),
+      variables: {
+        collection: new PublicKey(collectionKey),
+      },
+    }),
+  })
+  const body = await result.json()
+  const mintToCheck = body.data?.nfts[0]?.mintAddress
```

Comment:
`mintToCheck` may be undefined here. I think `new PublicKey(mintToCheck)` would fail.

Maybe throw a custom error to ease debugging if it happens? Or handle having no data if that's an accepted case